src/cloud/docker/docker-extend.md

```diff
@@ -143,9 +187,96 @@ To add a new PHP extension:
 
 1. Test the extension by specifying the [Docker build sources].
 
-[multiple compose files]: https://docs.docker.com/compose/reference/overview/#specifying-multiple-compose-files
-[service versions]: https://devdocs.magento.com/cloud/docker/docker-containers.html#service-containers
+### PHP extension configuration reference
+
+Use the following attributes to specify the PHP extension configuration in the getConfig method located in the [ExtensionResolver.php] file. The configuration you specify depends the extension installation method: from the official Docker PHP images, from the PECL repository, or using an installation script.
+
+| Configuration option | Description
+| -------------------- | ------------
+| PHP version constraint | Specifies the extension versions to install. If different versions have different installation requirements, you must add the configuration for each version.
+| `EXTENSION_TYPE_CORE` | Extension that can be installed from a `docker-php-source` image. See [PHP, Docker Official Images] in Docker Hub.
+| `EXTENSION_TYPE_PECL` | Extensions that can be installed from the [PECL] repository. See _PECL extensions_ in the [How to install more PHP extensions][PHP, Docker Official Images] documentation in Docker Hub.
+| `EXTENSION_TYPE_INSTALLATION_SCRIPT` | For extensions that install using a command sequence. See the _Other extensions_ section on the [PHP, Docker Official Images] page in Docker Hub.
+| `EXTENSION_TYPE` | Specifies whether the extension installed from the Docker PHP images, the PECL repository, or using an installation script. Valid values: `EXTENSION_TYPE_CORE`, `EXTENSION_TYPE_PECL`, or `EXTENSION_TYPE_INSTALLATION_SCRIPT`<br> For details on core and PECL extensions, see [How to install more PHP extensions][PHP, Docker Official Images] in Docker Hub.
+`EXTENSION_OS_DEPENDENCIES` | For PHP core or PECL extensions, specifies Linux package dependencies. These packages install in the order listed before installing the extension.
+`EXTENSION_CONFIGURE_OPTIONS` | For PHP core extensions, specifies any configuration options to apply when Docker configures the PHP extension using the `docker-php-ext-configure` command.  See the _PHP Core Extensions_ section on the [PHP, Docker Official Images] page in Docker Hub.
+`EXTENSION_PACKAGE_NAME` | Specifies the extension package name. If this parameter is specified, then the value is used to generate the installation command.
+`EXTENSION_INSTALLATION_SCRIPT` | For extension type `EXTENSION_TYPE_INSTALLATION_SCRIPT`, specifies the Bash script to install the extension.
+
+#### Example: Core extension configuration
+
+The following example shows the configuration for adding the PHP core extension `gd` in the `ExtensionResolver.php` file.
+
+```php?start_inline=1
+public static function getConfig(): array
+...
+         'gd' => [
```

Comment:
There is a lot of white space here. If it is just an excerpt, would the formatting matter for the sake of seeing a more compact view without scroll? Would they want to copy this example code and insert it somewhere exactly with all that indentation?

Reply:
I corrected the indention on first example. These example are copied directly from the ExtensionResolver.php file, so yes, they can just copy and paste and edit based on the extension they need to add.